packages/server/src/arpa_reporter/services/generate-arpa-report.js

```diff
@@ -945,68 +967,83 @@ async function setCSVData(data) {
 
-    const admZip = new AdmZip();
+        // generate every csv file for the report
+        const csvObjects = [
+            { name: 'project111210BulkUpload', func: generateProject111210 },
+            { name: 'project18_229233BulkUpload', func: generateProject18 },
+            { name: 'project19_234BulkUpload', func: generateProject19 },
+            { name: 'project211214BulkUpload', func: generateProject211214 },
+            { name: 'project2128BulkUpload', func: generateProject2128 },
+            { name: 'project215218BulkUpload', func: generateProject215218 },
+            { name: 'project224227BulkUpload', func: generateProject224227 },
+            { name: 'project236BulkUpload', func: generateProject236 },
+            { name: 'project31BulkUpload', func: generateProject31 },
+            { name: 'project32BulkUpload', func: generateProject32 },
+            { name: 'project4142BulkUpload', func: generateProject4142 },
+            { name: 'project51518BulkUpload', func: generateProject51518 },
+            { name: 'project519521BulkUpload', func: generateProject519521 },
+            { name: 'projectBaselineBulkUploadTemplate', func: generateProjectBaseline },
+            {
+                name: 'expendituresGT50000BulkUpload',
+                func: generateExpendituresGT50000,
+            },
+            {
+                name: 'expendituresLT50000BulkUpload',
+                func: generateExpendituresLT50000,
+            },
+            {
+                name: 'paymentsIndividualsLT50000BulkUpload',
+                func: generatePaymentsIndividualsLT50000,
+            },
+            { name: 'subawardBulkUpload', func: generateSubaward },
+            { name: 'subRecipientBulkUpload', func: generateSubRecipient },
+        ];
 
-    const reportName = await generateReportName(periodId, tenantId);
+        const admZip = new AdmZip();
 
-    // compute the CSV data for each file, and write it into the zip container
-    const inputs = [];
-    csvObjects.forEach((c) => inputs.push({
-        csvObject: c, admZip, records, periodId,
-    }));
-    await asyncBatch(inputs, setCSVData, 2);
+        const reportName = await generateReportName(periodId, tenantId);
 
-    // return the correct format
-    return {
-        filename: `${reportName}.zip`,
-        content: admZip.toBuffer(),
-    };
+        // compute the CSV data for each file, and write it into the zip container
+        for (const csvObject of csvObjects) {
+            /* here we want to evaluate each csvObject in series */
+            // eslint-disable-next-line no-await-in-loop
+            tracer.trace('setCSVData()', async () => {
+                logger.info({ csv: { name: csvObject.name } }, 'generating csv');
+                await setCSVData({
+                    csvObject, admZip, records, periodId,
+                }, logger.child({ csvObject: { name: csvObject.name } }));
+            });
+        }
```

Comment:
Setting `tracer.trace('setCSVData()', async () => {` is going to generate a bunch of traces all named `setCSVData()`, which might not be very helpful for debugging.

Recommendation:
```suggestion
        tracer.trace('setCSVData', async () => {
            for (const csvObject of csvObjects) {
                /* here we want to evaluate each csvObject in series */
                // eslint-disable-next-line no-await-in-loop
                tracer.trace(csvObject.name, async () => {
                    logger.info(`Generating ${csvObject.name} CSV`);
                    await setCSVData({
                        csvObject, admZip, records, periodId,
                    }, logger.child({ csvObject: { name: csvObject.name } }));
                });
            }
        });
```